python/cudf/cudf/core/frame.py

```diff
@@ -3812,3 +3813,37 @@ def _is_series(obj):
     instead of checking for isinstance(obj, cudf.Series)
     """
     return isinstance(obj, Frame) and obj.ndim == 1 and obj._index is not None
+
+
+def _drop_rows_by_labels(
+    obj: Union[cudf.DataFrame, cudf.Series],
+    labels: Union[ColumnLike, abc.Iterable, str],
+    level: Union[int, str],
+    errors: str,
+) -> Union[cudf.DataFrame, cudf.Series]:
+    """Remove rows specified by `labels`. If `errors=True`, an error is raised
+    if some items in `labels` do not exist in `obj._index`.
+
+    Parameter `level` is ignored if `obj._index` is not `MultiIndex`
+    """
+    if not isinstance(labels, (cudf.Series, cudf.Index)):
+        labels = as_column(labels)
+
+    if isinstance(obj._index, cudf.MultiIndex):
+        if level is None:
+            level = 0
+
+        levels_index = obj.index.get_level_values(level)
+        if errors == "raise" and not labels.isin(levels_index).all():
+            raise KeyError("One or more values not found in axis")
+
+        # TODO : Could use anti-join as a future optimization
+        sliced_df = obj.take(~levels_index.isin(labels))
+        sliced_df._index.names = obj._index.names
```

Comment:
Is it possible/more efficient to use an anti-join on `obj._index._source_data` and then reconstruct a `MultiIndex` from the result?

Reply:
I think so, the general idea is: create an intermediate dataframe with the selected `level` from the midx as the index. The dataframe has the rest columns of `._index._source_data` and `Series`/`DataFrame` values, then do a "leftanti` join.